DentOS_Framework/DentOsTestbedLib/src/dent_os_testbed/lib/traffic/ixnetwork/ixnetwork_ixia_client_impl.py

```diff
@@ -663,3 +666,42 @@ def run_send_arp(self, device, command, *argv, **kwarg):
         return err, [{"success": msg["arg2"],
                       "port": msg["port"],
                       "src_ip": msg["src_ip"]} for msg in res]
+
+    def run_update_l1_config(self, device, command, *argv, **kwarg):
+        if not IxnetworkIxiaClientImpl.ixnet:
+            return 0, "Ixia not connected"
+        if command == "update_l1_config":
+            vports = IxnetworkIxiaClientImpl.ixnet.Vport.find()
+            ports = kwarg['params'][0].get("ixia_ports", [])
+            if len(ports) < 1:
+                raise ValueError(f" IXIA ports not provided")
+            # fd -> full duplex ; hd -> half duplex
+            duplex = "fd" if kwarg['params'][0].get("duplex", "Full").capitalize() == "Full" else "hd"
```

Comment:
Half duplex is default by the standard, but only for 10/100. Maybe it's better to follow the same approach here as well. For 1G/10G without providing duplex explicitly should be "Full".

Reply:
IXIA speeds are in three different formats `speed10fd, speed100hd, speed10g ` (endings on **hd, fd , g**)To convert user provided speed to ixia supported speed  use  `__convert_to_ixia_speed(self, speed, duplex)` function